backend/auth/role_map_repository.go

```diff
@@ -56,6 +56,38 @@ func (rmr *RoleMapRepository) HasPermission(rolenames []string, operation *model
 	return false
 }
 
+func (rmr *RoleMapRepository) GetAllPermissions(role []string) map[string]map[string]map[models.OperationType]struct{} {
+	pmatrix := make(map[string]map[string]map[models.OperationType]struct{})
+	first := true
+	for _, r := range role {
+		if _, exists := rmr.flattenedMap[r]; exists {
+			if first {
+				pmatrix = deepCopy(rmr.flattenedMap[r])
+				pruneResourcesNamespaces(pmatrix)
```

Comment:
Honestly I have no idea. The basic idea is that it will keep the matrix from growing to larger sizes and thus increasing the cost of each operation. On the other hand if we prune some resources/namespaces that will have to be expanded again into in next step than we actually waste some time. I was thinking about doing it once every n operations or tracking the set of leftover namespaces/resources but it would require some extra work which is maybe not worth that much

Reply:
If you have any clever idea I am open for suggestions

Reply:
I think doing it at the end will be enough, but it can stay